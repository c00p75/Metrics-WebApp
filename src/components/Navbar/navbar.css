a {
  color: white !important;
}

nav {
  position: sticky;
  z-index: 2;
  top: -1px;
  left: 0;
  color: white;
  background-color: var(--primary-color);
}

nav button {
  font-size: 2em;
  color: white;
}

nav span {
  font-weight: 500;
}

.menu {
  display: flex;
  justify-content: center;
  align-items: center;
}

.App__navbar-searchContainer {
  position: relative;

  --size-button: 40px;
}

.App__navbar-searchContainer .icon {
  position: absolute;
  width: var(--size-button);
  height: var(--size-button);
  top: 0;
  left: 0;
  padding: 8px;
  pointer-events: none;
}

.App__navbar-searchContainer .icon svg {
  width: 100%;
  height: 100%;
}

.input {
  padding-left: var(--size-button);
  height: var(--size-button);
  font-size: 1em;
  border: none;
  color: black;
  outline: none;
  width: var(--size-button);
  background: none;
  transition: all ease 0.3s;
  cursor: pointer;
}

.input:focus {
  width: 100%;
  cursor: text;
  background-color: whitesmoke;
  border-radius: 20px;
}

.input:focus + .icon {
  pointer-events: all;
  cursor: pointer;
}

.activeRoute {
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  font-size: 1.3em;
}
