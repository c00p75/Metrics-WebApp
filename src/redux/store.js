import { configureStore } from '@reduxjs/toolkit';
import countriesReducer from './countries/countrySlice';

const store = configureStore({
  reducer: {
    countries: countriesReducer,
  },
});

export default store;
