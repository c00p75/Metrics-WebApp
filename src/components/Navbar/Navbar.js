import React, { useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { BiMenu } from 'react-icons/bi';
import { useNavigate } from 'react-router-dom';
import './navbar.css';
import { searchFocus } from '../../redux/countries/countrySlice';

const Navbar = () => {
  const dispatch = useDispatch();
  const countries = useSelector((state) => state.countries);
  const navigate = useNavigate();
  const { currentPage } = countries;
  const [inputFocus, setInputFocus] = useState(false);
  const [, setInput] = useState('');
  const handleInput = (e) => {
    setInput(e.target.value);
    dispatch(searchFocus(e.target.value));
  };

  const clearSearch = () => {
    setInput('');
    dispatch(searchFocus(''));
    if (document.querySelector('input')) document.querySelector('input').value = '';
  };

  const location = window.location.pathname;
  let path = location;
  if (path.includes('%20')) path = path.replace(/%20/g, ' ');

  return (
    <nav className="d-flex justify-content-between align-items-center p-2">
      {(path === '/' && !inputFocus) && (
        <>
          <BiMenu style={{ fontSize: '2em' }} />
          <span className="activeRoute mx-3">World Population</span>
        </>
      )}

      {(path !== '/' && !inputFocus) && (
        <>
          <button type="button" style={{ background: 'none', border: 'none' }} onClick={() => { clearSearch(); navigate(-1); }}>‚ùÆ</button>
          <span className="activeRoute px-4">
            {(currentPage !== 'home' && currentPage !== 'continents') ? `${path.split('/')[1]}/${currentPage}` : path.slice(1, path.length)}
          </span>

        </>
      )}

      <div className={(currentPage !== 'home' && currentPage !== 'continents') ? 'd-none' : 'App__navbar-searchContainer'} style={{ width: inputFocus ? '100%' : '' }}>
        <input onFocus={() => setInputFocus(true)} onBlur={() => { setInputFocus(false); clearSearch(); }} onChange={(e) => handleInput(e)} placeholder={path === '/' ? 'Search all...' : `Search ${path.slice(1, path.length)}...`} required="" className="input" name="text" type="text" />
        <div className="icon">
          <svg viewBox="0 0 512 512" className="ionicon" xmlns="http://www.w3.org/2000/svg">
            <title>Search</title>
            <path strokeWidth="32" strokeMiterlimit="10" stroke="currentColor" fill="none" d="M221.09 64a157.09 157.09 0 10157.09 157.09A157.1 157.1 0 00221.09 64z" />
            <path d="M338.29 338.29L448 448" strokeWidth="32" strokeMiterlimit="10" strokeLinecap="round" stroke="currentColor" fill="none" />
          </svg>
        </div>
      </div>
    </nav>
  );
};

export default Navbar;
