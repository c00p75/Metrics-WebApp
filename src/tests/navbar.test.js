import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import Navbar from '../components/Navbar/Navbar';

it('matches snapshot', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <Navbar />
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
