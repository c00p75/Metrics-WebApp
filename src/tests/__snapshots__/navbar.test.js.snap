// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<nav
  className="d-flex justify-content-between align-items-center p-2"
>
  <button
    onClick={[Function]}
    style={
      Object {
        "background": "none",
        "border": "none",
      }
    }
    type="button"
  >
    f
  </button>
  <span>
    Continents
  </span>
  <div
    className="App__navbar-searchContainer"
  >
    <input
      className="input"
      name="text"
      onChange={[Function]}
      placeholder="Search all..."
      required=""
      type="text"
    />
    <div
      className="icon"
    >
      <svg
        className="ionicon"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          Search
        </title>
        <path
          d="M221.09 64a157.09 157.09 0 10157.09 157.09A157.1 157.1 0 00221.09 64z"
          fill="none"
          stroke="currentColor"
          strokeMiterlimit="10"
          strokeWidth="32"
        />
        <path
          d="M338.29 338.29L448 448"
          fill="none"
          stroke="currentColor"
          strokeLinecap="round"
          strokeMiterlimit="10"
          strokeWidth="32"
        />
      </svg>
    </div>
  </div>
  <div
    className="d-flex justify-content-between"
    style={
      Object {
        "width": "20%",
      }
    }
  >
    <button
      className="menu"
      onClick={[Function]}
      style={
        Object {
          "background": "none",
          "border": "none",
        }
      }
      type="button"
    >
      <svg
        className="bi bi-mic-fill"
        fill="white"
        height="20"
        viewBox="0 0 16 16"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M5 3a3 3 0 0 1 6 0v5a3 3 0 0 1-6 0V3z"
        />
        <path
          d="M3.5 6.5A.5.5 0 0 1 4 7v1a4 4 0 0 0 8 0V7a.5.5 0 0 1 1 0v1a5 5 0 0 1-4.5 4.975V15h3a.5.5 0 0 1 0 1h-7a.5.5 0 0 1 0-1h3v-2.025A5 5 0 0 1 3 8V7a.5.5 0 0 1 .5-.5z"
        />
      </svg>
    </button>
    <button
      className="menu"
      onClick={[Function]}
      style={
        Object {
          "background": "none",
          "border": "none",
        }
      }
      type="button"
    >
      <svg
        className="bi bi-gear-wide"
        fill="currentColor"
        height="16"
        viewBox="0 0 16 16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8.932.727c-.243-.97-1.62-.97-1.864 0l-.071.286a.96.96 0 0 1-1.622.434l-.205-.211c-.695-.719-1.888-.03-1.613.931l.08.284a.96.96 0 0 1-1.186 1.187l-.284-.081c-.96-.275-1.65.918-.931 1.613l.211.205a.96.96 0 0 1-.434 1.622l-.286.071c-.97.243-.97 1.62 0 1.864l.286.071a.96.96 0 0 1 .434 1.622l-.211.205c-.719.695-.03 1.888.931 1.613l.284-.08a.96.96 0 0 1 1.187 1.187l-.081.283c-.275.96.918 1.65 1.613.931l.205-.211a.96.96 0 0 1 1.622.434l.071.286c.243.97 1.62.97 1.864 0l.071-.286a.96.96 0 0 1 1.622-.434l.205.211c.695.719 1.888.03 1.613-.931l-.08-.284a.96.96 0 0 1 1.187-1.187l.283.081c.96.275 1.65-.918.931-1.613l-.211-.205a.96.96 0 0 1 .434-1.622l.286-.071c.97-.243.97-1.62 0-1.864l-.286-.071a.96.96 0 0 1-.434-1.622l.211-.205c.719-.695.03-1.888-.931-1.613l-.284.08a.96.96 0 0 1-1.187-1.186l.081-.284c.275-.96-.918-1.65-1.613-.931l-.205.211a.96.96 0 0 1-1.622-.434L8.932.727zM8 12.997a4.998 4.998 0 1 1 0-9.995 4.998 4.998 0 0 1 0 9.996z"
        />
      </svg>
    </button>
  </div>
</nav>
`;
